code/modules/reagents/Chemistry-Holder.dm

```diff
@@ -23,6 +23,15 @@ datum
 				for(var/path in paths)
 					var/datum/reagent/D = new path()
 					chemical_reagents_list[D.id] = D
+			if(!safe_reagents_list)
+				//Filtered list to prevent shit like xenobotany spitting out 200 bar drinks
+				var/paths = typesof(/datum/reagent) - /datum/reagent
+				safe_reagents_list = list()
+				for(var/path in paths)
+					var/datum/reagent/D = new path()

```

Comment:
okay so either I'm doing this horrifyingly wrong or this doesn't work but trying this results in "tried to pick from empty list"

i tried 3 different methods, all of them failed, i think this is the only way to go